client/src/SignUpShopper.js

```diff
@@ -0,0 +1,98 @@
+import { useEffect, useState } from 'react';
+import { useHistory, useParams } from 'react-router-dom';
+import Api from './Api';
+
+//to add new component, make new file
+//to create (crud), use forms
+function SignUpShopper() {
+
+    //:id from Sections.js 
+    const {id} = useParams();
+    const history = useHistory();
+    
+    //initial value of section (in this case, obj with name, slug, and position attributes)
+    const [shopper, setShopper] = useState({
+        UserId: 0,
+        zipcode: 0
```

Comment:
Missing the bio here. Also, similar to the previous comment about not needing the user id in the route path, I don't think we'll need it here either.

User and Shopper are in a ONE-TO-ONE relationship, I imagine, correct? So if you know you're logged in as a given user, the server should then be able to find the ONE unique shopper record for that user (or create it if necessary) without having to bother with the ids in this case...

This is something we can take advantage of specifically with one-to-one relationships with the logged-in user.